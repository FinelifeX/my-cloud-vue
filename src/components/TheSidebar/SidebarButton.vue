<template>
  <button class="sidebar-button" @click="$emit('click')">
    <component :is="icon"></component>
    <span class="sidebar-button__text">{{ text }}</span>
  </button>
</template>

<script>
  export default {
    name: 'SidebarButton',
    props: {
      text: {
        type: String,
        required: true,
      },
      icon: {
        type: Object,
        default: null,
      },
    },
  };
</script>

<style scoped lang="scss">
  .sidebar-button {
    @apply w-full;
    @apply flex flex-row items-center;
    @apply text-spun_pearl;
    @apply transition-colors duration-150;
    @apply pt-3 pb-3 pl-4 pr-4;
    @apply rounded-lg;

    &:hover,
    &:target,
    &:focus {
      @apply text-indigo-600;
    }

    &:active {
      @apply bg-selago;
    }
  }

  .sidebar-button__text {
    @apply ml-6;
  }
</style>
