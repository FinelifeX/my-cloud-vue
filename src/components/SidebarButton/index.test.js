import { render } from '@testing-library/vue';
import SidebarButton from './index';

describe('SidebarButton component', () => {
  it('should render correctly', () => {
    const { html } = render(SidebarButton, { props: { text: 'button' } });

    expect(html()).toMatchSnapshot();
  });
});
