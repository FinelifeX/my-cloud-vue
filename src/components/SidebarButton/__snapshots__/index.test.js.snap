// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarButton component should render correctly 1`] = `
"<button class=\\"sidebar-button\\">
  <!----> <span class=\\"sidebar-button__text\\">button</span></button>"
`;
